### PR TITLE
fix: Refetch recent tickets

### PR DESCRIPTION
**Pull request by gorandalum**
By using fare contracts as dependency for the hook fetching recent
tickets they are updated when logged in user is changed and when new
tickets are bought.

### CODE REVIEW

File: src/screens/Ticketing/Tickets/use-recent-tickets.ts [hunk 1]
```diff
@@ -207,7 +212,7 @@ export default function useRecentTickets() {
 
   useEffect(() => {
     fetchRecentFareContracts();
-  }, []);
+  }, [fareContracts]);
```
Comment (cbrevik) [inline src/screens/Ticketing/Tickets/use-recent-tickets.ts:215]: Suggest changing to array-length, so we don't get unecessary updates

```suggestion
  }, [fareContracts.length]);
```

**Review by cbrevik — CHANGES_REQUESTED** [review]
I'm a bit unsure under what circumstances this will change. Will the array change if for example a ticket gets its QR-code updated? To avoid unnecessary API-calls, maybe we want to only update if the number of fare contracts changes?

File: src/screens/Ticketing/Tickets/use-recent-tickets.ts (hunk 1 above)
Comment (gorandalum) [reply src/screens/Ticketing/Tickets/use-recent-tickets.ts:215]: Good point. Maybe we could use `[fareContracts[0]?.id]` to also eliminate the (very few) cases where logging in/out to a user with equal ticket count?

Comment (cbrevik) [reply src/screens/Ticketing/Tickets/use-recent-tickets.ts:215]: Hmm, good point! 🤔 That's probably a good idea. Another way (so the way the array sort would not matter) would be to evaluate the latest `created` time. Something like `Math.max(0, ...fareContracts.map((a) => a.created.seconds));` But not sure if we're over-analyzing the whole thing then?

**Pull request merged**